test/python/circuit/test_circuit_properties.py

```diff
@@ -1267,6 +1267,18 @@ def test_metadata_copy_does_not_share_state(self):
 
         self.assertEqual(qc1.metadata["a"], 0)
 
+    def test_metadata_is_dict(self):
+        """Verify setting metadata to None in the constructor results in an empty dict."""
+        qc = QuantumCircuit(1)
+        metadata1 = qc.metadata
+        self.assertEqual(metadata1, {})
+
+    def test_metadata_raises(self):
+        """Test that we must set metadata to a dict."""
+        qc = QuantumCircuit(1)
+        with self.assertRaises(TypeError):
+            qc.metadata = 1
```

Comment:
we also like to test that any deprecation warnings fire off and perform the correct action as well, so could we also have a
```python
qc = QuantumCircuit(1)
with self.assertWarns(DeprecationWarning):
    qc.metadata = None
self.assertEqual(qc.metadata, {})
```
case?

Reply:
092fb4b3c3 :+1: